cmake/Modules/FindComputeCpp.cmake

```diff
@@ -292,33 +292,36 @@ function(__build_spir targetName sourceFile binaryDir fileCounter)
   # Add a dependency on the integration header
   add_dependencies(${targetName} ${headerTargetName})
 
+  # This property can be set on a per-target basis to indicate that the
+  # integration header should appear after the main source listing
+  get_property(includeAfter TARGET ${targetName}
+      PROPERTY COMPUTECPP_INCLUDE_AFTER)
+
+  if(includeAfter)
+    # Change the source file to the integration header - e.g.
+    # g++ -c source_file_name.cpp.sycl
+    set_property(TARGET ${targetName} PROPERTY SOURCES ${outputSyclFile})
+    # CMake/gcc don't know what language a .sycl file is, so tell them
+    set_property(SOURCE ${outputSyclFile} PROPERTY LANGUAGE CXX)
+    set(includedFile ${sourceFile})
+  else()
+    set(includedFile ${outputSyclFile})
+  endif()
+
   # Force inclusion of the integration header for the host compiler
   if(MSVC)
     # NOTE: The Visual Studio generators parse compile flags differently,
     # hence the different argument syntax
     if(CMAKE_GENERATOR MATCHES "Visual Studio")
-      set(forceIncludeFlags "/FI\"${outputSyclFile}\"")
+      set(forceIncludeFlags "/FI\"${includedFile}\" /TP")
     else()
-      set(forceIncludeFlags /FI ${outputSyclFile})
+      set(forceIncludeFlags /FI ${includedFile} /TP)
```

Comment:
What is this else for? ninja?

Reply:
I believe so. This functionality was added by Máté, he left that comment saying that the VS generator works differently. I've never encountered it personally but my CMake + Windows experience is rather more limited

Reply:
It is a rather interesting situation, since in theory all properties are lists. I wonder if the underlying reason for using strings on the visual studio generator is because it was removing duplicated flags? If not, I think it should be a bug in CMake, since properties should be agnostics to targets

Reply:
It might be a CMake bug! We could test both ninja and VS with both versions to see if there's some common behaviour so we can settle on one or the other, but I'm not set up to do that here. This might be best placed in a different PR - could you make an issue for this?

Reply:
https://github.com/codeplaysoftware/computecpp-sdk/issues/79  done!